hazelcast/src/main/java/com/hazelcast/client/impl/connection/ClientConnectionManager.java

```diff
@@ -95,4 +95,10 @@ public interface ClientConnectionManager extends ConnectionListenable<ClientConn
      * the TPC config is disabled.
      */
     boolean isUnisocketClient();
+
+    /**
+     * Returns {@code true} if the client is connected to a cluster with
+     * enterprise features enabled, {@code false} otherwise.
+     */
+    boolean isConnectedToEnterpriseCluster();
```

Comment:
Why do we have this on connection manager? It is a cluster information and I expect it to be on the cluster service instead, right? What do you think?

Reply:
The main method that is public is in Cluster.java. This isConnectedToEnterpriseCluster is not a public method, it is used by ClientClusterProxy (Cluster implementor).

The reason I put it in the connection manager is because the connection manager authenticates with the cluster. In the authentication response we have the information we use: isFailoverSupported.

Basically, the conn. manager keeps track of the last connected cluster's isFailoverSupported value in isConnectedToEnterpriseCluster variable and returns it via this method.